File: exams/frmClassSubjects.vb
Public Class frmClassSubjects

    Private Sub frmClassSubjects_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            get_subjects()
            cboSubject.Items.Clear()

            For k As Integer = 0 To subjabb.Length - 1
                cboSubject.Items.Add(subjects(k))
            Next
            load_class(cboClass)
        End If
    End Sub

    Private Sub cboClass_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cboClass.SelectedIndexChanged
        lstSubjects.Items.Clear()
        Dim i As Integer = 1
        qread("SELECT * FROM class_subjects WHERE class='" & escape_string(cboClass.SelectedItem) & "'")
        While dbreader.Read
            Dim li As New ListViewItem
            li = lstSubjects.Items.Add(i)
            li.Tag = dbreader("SubjID")
            li.SubItems.Add(dbreader("Class"))
            li.SubItems.Add(dbreader("Subject"))
            li.SubItems.Add(dbreader("Abbreviation"))
            li.SubItems.Add(dbreader("Department"))
            li.SubItems.Add(dbreader("Comment"))
            li.SubItems.Add(dbreader("Code"))
            i += 1
        End While
    End Sub

    Private Function Exists() As Boolean
        For k As Integer = 0 To lstSubjects.Items.Count - 1
            If cboSubject.SelectedItem = lstSubjects.Items.Item(k).SubItems(2).Text Then
                Return True
            End If
        Next
        Return False
    End Function

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        start()
        qwrite("DELETE FROM class_subjects WHERE class='" & escape_string(cboClass.SelectedItem) & "'")
        For k As Integer = 0 To lstSubjects.Items.Count - 1
            If Not qwrite("INSERT INTO class_subjects VALUES(NULL,'" & escape_string(lstSubjects.Items.Item(k).SubItems(1).Text) & "','" & escape_string(lstSubjects.Items.Item(k).SubItems(2).Text) & "','" & escape_string(lstSubjects.Items.Item(k).SubItems(3).Text) & "','" & escape_string(lstSubjects.Items.Item(k).SubItems(4).Text) & "','" & escape_string(lstSubjects.Items.Item(k).SubItems(5).Text) & "','" & escape_string(lstSubjects.Items.Item(k).SubItems(6).Text) & "')") Then
                rollback()
                failure("Could Not Save Class Subjects!")
                Exit Sub
            End If
        Next

        commit()
        success("Class Subjects Successfully Saved!")
    End Sub



    Private Sub Button5_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button5.Click
        If lstSubjects.SelectedItems.Count > 0 Then
            If confirm("Are you sure you want to delete the selected subject from the list?") Then
                For k As Integer = lstSubjects.Items.Count - 1 To 0 Step -1
                    If lstSubjects.Items.Item(k).Selected Then
                        lstSubjects.Items.RemoveAt(k)
                    End If
                Next
            End If
        End If
    End Sub

    Private Sub SimpleButton1_Click(sender As Object, e As EventArgs) Handles SimpleButton1.Click
        If Not Exists() Then
            qread("SELECT * FROM subjects WHERE subject='" & escape_string(cboSubject.SelectedItem) & "'")
            dbreader.Read()
            Dim li As New ListViewItem
            li = lstSubjects.Items.Add(dbreader("ID"))
            li.SubItems.Add(cboClass.SelectedItem)
            li.SubItems.Add(dbreader("Subject"))
            li.SubItems.Add(dbreader("Abbreviation"))
            li.SubItems.Add(dbreader("Department"))
            li.SubItems.Add(dbreader("Comment"))
            ' previous code li.SubItems.Add(dbreader("Code"))
            li.SubItems.Add(dbreader("subjectCode"))
        Else
            failure("Subject Alread Exists In The List!")
        End If
    End Sub
End Class

File: exams/frmClassSubjects.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmClassSubjects
    {
        public frmClassSubjects()
        {
            InitializeComponent();
            _Button5.Name = "Button5";
            _Button2.Name = "Button2";
            _SimpleButton1.Name = "SimpleButton1";
            _cboClass.Name = "cboClass";
        }

        private void frmClassSubjects_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                publicSubsNFunctions.get_subjects();
                cboSubject.Items.Clear();
                for (int k = 0, loopTo = publicSubsNFunctions.subjabb.Length - 1; k <= loopTo; k++)
                    cboSubject.Items.Add(publicSubsNFunctions.subjects[k]);
                var argcbo = cboClass;
                publicSubsNFunctions.load_class(ref argcbo);
                cboClass = argcbo;
            }
        }

        private void cboClass_SelectedIndexChanged(object sender, EventArgs e)
        {
            lstSubjects.Items.Clear();
            int i = 1;
            string argq = "SELECT * FROM class_subjects WHERE class='" + publicSubsNFunctions.escape_string(Conversions.ToString(cboClass.SelectedItem)) + "'";
            publicSubsNFunctions.qread(ref argq);
            while (publicSubsNFunctions.dbreader.Read())
            {
                var li = new ListViewItem();
                li = lstSubjects.Items.Add(i.ToString());
                li.Tag = publicSubsNFunctions.dbreader["SubjID"];
                li.SubItems.Add(publicSubsNFunctions.dbreader["Class"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Subject"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Abbreviation"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Department"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Comment"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Code"]);
                i += 1;
            }
        }

        private bool Exists()
        {
            for (int k = 0, loopTo = lstSubjects.Items.Count - 1; k <= loopTo; k++)
            {
                if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(cboSubject.SelectedItem, lstSubjects.Items[k].SubItems[2].Text, false)))
                {
                    return true;
                }
            }

            return false;
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            publicSubsNFunctions.start();
            publicSubsNFunctions.qwrite("DELETE FROM class_subjects WHERE class='" + publicSubsNFunctions.escape_string(Conversions.ToString(cboClass.SelectedItem)) + "'");
            for (int k = 0, loopTo = lstSubjects.Items.Count - 1; k <= loopTo; k++)
            {
                if (!publicSubsNFunctions.qwrite("INSERT INTO class_subjects VALUES(NULL,'" + publicSubsNFunctions.escape_string(lstSubjects.Items[k].SubItems[1].Text) + "','" + publicSubsNFunctions.escape_string(lstSubjects.Items[k].SubItems[2].Text) + "','" + publicSubsNFunctions.escape_string(lstSubjects.Items[k].SubItems[3].Text) + "','" + publicSubsNFunctions.escape_string(lstSubjects.Items[k].SubItems[4].Text) + "','" + publicSubsNFunctions.escape_string(lstSubjects.Items[k].SubItems[5].Text) + "','" + publicSubsNFunctions.escape_string(lstSubjects.Items[k].SubItems[6].Text) + "')"))
                {
                    publicSubsNFunctions.rollback();
                    publicSubsNFunctions.failure("Could Not Save Class Subjects!");
                    return;
                }
            }

            publicSubsNFunctions.commit();
            publicSubsNFunctions.success("Class Subjects Successfully Saved!");
        }

        private void Button5_Click(object sender, EventArgs e)
        {
            if (lstSubjects.SelectedItems.Count > 0)
            {
                if (Conversions.ToBoolean(publicSubsNFunctions.confirm("Are you sure you want to delete the selected subject from the list?")))
                {
                    for (int k = lstSubjects.Items.Count - 1; k >= 0; k -= 1)
                    {
                        if (lstSubjects.Items[k].Selected)
                        {
                            lstSubjects.Items.RemoveAt(k);
                        }
                    }
                }
            }
        }

        private void SimpleButton1_Click(object sender, EventArgs e)
        {
            if (!Exists())
            {
                string argq = "SELECT * FROM subjects WHERE subject='" + publicSubsNFunctions.escape_string(Conversions.ToString(cboSubject.SelectedItem)) + "'";
                publicSubsNFunctions.qread(ref argq);
                publicSubsNFunctions.dbreader.Read();
                var li = new ListViewItem();
                li = lstSubjects.Items.Add(publicSubsNFunctions.dbreader["ID"]);
                li.SubItems.Add(cboClass.SelectedItem);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Subject"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Abbreviation"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Department"]);
                li.SubItems.Add(publicSubsNFunctions.dbreader["Comment"]);
                // previous code li.SubItems.Add(dbreader("Code"))
                li.SubItems.Add(publicSubsNFunctions.dbreader["subjectCode"]);
            }
            else
            {
                publicSubsNFunctions.failure("Subject Alread Exists In The List!");
            }
        }
    }
}